### PR TITLE
level-up

### PR DESCRIPTION
**Pull request by diasdavid**
This PR was about making the generic commands interface-ipfs-core compliant, however, I learned that with the use of the latest ipfs-api (on a branch at a time, now released), that our http-api tests are very brittle, so I decided to make this PR something more in order to make js-ipfs-api more robust.

- [x] generic api compliant
- [x] update to ipfs-api 7.0.0
- [x] refactor and clean up http-api tests

### CODE REVIEW

**Comment by diasdavid** [comment]
@dignifiedquire @haadcode @JGAntunes @nginnever @shamb0t @VictorBjelkholm @RichardLitt wanna CR this one? :)

File: package.json [hunk 1]
```diff
@@ -67,8 +67,9 @@
     "fs-blob-store": "^5.2.1",
     "glob": "^7.0.5",
     "hapi": "^14.0.0",
+    "ipfs-api": "^7.0.0",
     "ipfs-bitswap": "^0.6.0",
-    "ipfs-api": "^6.0.3",
+    "ipfs-api": "^7.0.0",
```
Comment (JGAntunes) [inline package.json]: @diasdavid this dep is repeated.

**Comment by diasdavid** [comment]
@dignifiedquire  have you seen this:

```
PhantomJS 2.1.1 (Linux 0.0.0) ERROR
  TypeError: undefined is not a function (evaluating 'Array.from(arguments)')
  at webpack:///~/mafmt/lib/index.js:66:0 <- test/browser.js:18044
```

Getting it on CI and locally with the new Aegir

File: test/cli/test-config.js [hunk 2]
```diff
@@ -132,7 +132,7 @@ describe('config', () => {
     })
 
     describe('get/set', () => {
-      it('get a config key value', (done) => {
```
Comment (diasdavid) [inline test/cli/test-config.js]: will tackle all of the CLI tests after http-api compliance with https://github.com/ipfs/js-ipfs/issues/428

File: src/core/ipfs/config.js [hunk 3]
```diff
@@ -1,14 +1,56 @@
...
+        return callback(new Error('Invalid key type'))
+      }
+
+      self._repo.config.get((err, config) => {
+        if (err) {
+          return callback(err)
+        }
+        const value = _get(config, key, undefined)
+        if (!value) {
+          callback(new Error('Key does not exist in config'))
```
Comment (dignifiedquire) [inline src/core/ipfs/config.js]: This is not entirely accurate, `value` could be an empty string for `0`, and would as such evaluate to `true` in this check.

File: src/core/ipfs/config.js [hunk 4]
```diff
@@ -1,14 +1,56 @@
...
+          callback(null, value)
+        }
+      })
+    }),
+    set: promisify((key, value, callback) => {
+      if (!key || typeof key !== 'string') {
+        return callback(new Error('Invalid key type'))
+      }
+
+      if (!value || Buffer.isBuffer(value)) {
```
Comment (dignifiedquire) [inline src/core/ipfs/config.js]: Same issue as above, can't set values to `0` or `''`

File: test/cli/test-swarm.js [hunk 5]
```diff
@@ -58,7 +58,7 @@ describe('swarm', function () {
         })
     })
 
-    it('peers', (done) => {
+    it.skip('peers', (done) => {
```
Comment (dignifiedquire) [inline test/cli/test-swarm.js]: why are these skipped?

File: src/core/ipfs/config.js (hunk 3 above)
Comment (diasdavid) [inline src/core/ipfs/config.js]: good catch!

File: test/cli/test-swarm.js (hunk 5 above)
Comment (diasdavid) [inline test/cli/test-swarm.js]: Same as: https://github.com/ipfs/js-ipfs/pull/420#discussion_r75692397

**Comment by diasdavid** [comment]
Applied the CR, going to wait for CI (node.js + browser) and open a issue about the aegir and the ES5 version. @dignifiedquire is already on top of it.

**Pull request merged**